querying/github.go

```diff
@@ -242,33 +165,59 @@ func (gh *GithubDataSource) gatherRepoOwners(projects *ProjectCollection) {
 
 	for {
 		log.Info().Msg("Querying GitHub API for repository ownership information.")
-		if err := gh.GhClient.Query(gh.ctx, &ownerQuery, queryVars); err != nil {
+		if err := gh.queryRepoOwners(&ownerQuery, queryVars); err != nil {
 			log.Fatal().Err(err).Msg("Failed to query GitHub for repository ownership.")
 		}
-		for _, team := range ownerQuery.Organization.Teams.Nodes {
-			teamConfig, err := configs.GetTeamConfigBySlug(team.Slug, gh.conf.Team)
-			if err != nil {
-				log.Warn().Err(err).Str("slug", team.Slug).Msg("Failed to load team from configs.")
+
+		gh.processRepoOwners(&ownerQuery, projects, log)
```

Comment:
Go lint suggested moving this into a function to make the gatherRepoOwners function more readable.